session.go

```diff
@@ -252,11 +257,20 @@ func (s *Session) RunAnalyzeQuery(stmt spanner.Statement) (*pb.QueryPlan, error)
 
 func (s *Session) runQueryWithOptions(stmt spanner.Statement, opts spanner.QueryOptions) (*spanner.RowIterator, *spanner.ReadOnlyTransaction) {
 	if s.InReadWriteTransaction() {
+		if s.tc.transactionTag != "" {
+			// In a read-write transaction, add transaction tags as request tags.
+			opts.RequestTag = s.tc.transactionTag
+		}
 		iter := s.tc.rwTxn.QueryWithOptions(s.ctx, stmt, opts)
 		s.tc.sendHeartbeat = true
 		return iter, nil
 	}
 	if s.InReadOnlyTransaction() {
+		if s.tc.requestTag != "" {
```

Comment:
Same to the above. We don't need this `if` guard.

Reply:
fixed.